### ISSUE TITLE
Offline Cache Not Clear

### ISSUE THREAD

**Issue by AmeerChand** [rs/SDWebImage#2410]
### New Issue Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [x] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | ios 
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.4.1(15G77)  / 8.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | cocoapods 
 Xcode Version           |  Xcode 9 / Xcode 10 beta
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

my Device was online i download all images
it's work fine
but the issue is with offline (airplane mode) Cache Clearing.
i am using the below code to clear Cache.but not working.
it show's me the actual cache  size on disk
after clear memory and clear disk
it show me 0 size.
but not clear the cache.my device is still offline (AirPlan Mode) 
i am scrolling the TableViewCellImage it's show all image while i already Clear the Cache in offline mode.

```swift
print("The size on disk is =", SDImageCache.shared().getSize())
SDImageCache.shared().clearMemory()
SDImageCache.shared().clearDisk()
print("The size on disk is =", SDImageCache.shared().getSize())
```

above code output is 
The size on disk is = 5310705
The size on disk is = 0
but not Clear the Cache?
what to do to Clear the Cache in offline (airplane mode)

**Comment by bpoplauschi** [comment]
@AmeerChand clearing the cache does not set the images of the image views to nil.
If the UI elements already have a UIImage stored, they will exist until reload.
Are you saying that even new cells are properly loading the images? Then we need to check the memory cache.
Please fill in the issue template - I added it back as it was originally when you created the issue

**Comment by AmeerChand** [comment]
@bpoplauschi I am showing images via SDWebImage in a tableview. Now I have a button that clears the cache and when I reload the tableview it still shows images. Even I have killed the application completely after clearing cache and device is in airplane mode. Now when I relaunch the application it still shows images which I had cleared earlier.  That's why I think there is some issue in clearing cache.

**Comment by dreampiggy** [comment]
Could you please provide a demo to trigger this issue ? I test with some demo (including the demo in SDWebImage repo), but this does not cause the issue.

I think maybe you do a secondary cache system in your own business code. So I want to ensure this issue is happend from our library side.. Thanks

**Comment by bpoplauschi** [comment]
Closing due to lack of response. @AmeerChand feel free to reopen/continue the conversation
